mirgecom/fluid.py

```diff
@@ -313,6 +312,30 @@ def join_conserved(dim, mass, energy, momentum,
     return result
 
 
+def create_conserved(dim, mass, energy, momentum, species_mass=None):
```

Comment:
Why is this needed? Couldn't we just call the `ConservedVars` constructor?

Reply:
Can we overload the constructor and retain the shape and structure validation?

Reply:
Fair point!

There's `__post_init__` in a dataclass ([docs](https://docs.python.org/3/library/dataclasses.html)), but revalidating on every object construction is an easy way to get really slow code. I'm warming to having an explicit function that does this. I've tended to call these `make_xxx`. (examples: `make_kernel` in loopy, `make_xxx` in [pytato](https://documen.tician.de/pytato/reference.html#user-facing-node-creation))

Reply:
I hope this helps a little. (d012fba)